src/test/integration/viewNavigation.test.ts

```diff
@@ -0,0 +1,38 @@
+import { strictEqual } from 'assert';
+import { configuration } from '../../snyk/configuration';
+import { getExtension } from '../../extension';
+import { SNYK_VIEW_FEATURES, SNYK_VIEW_WELCOME } from '../../snyk/constants/views';
+import { TreeView } from 'vscode';
+import { Node } from '../../snyk/view/node';
+import { FeaturesViewProvider } from '../../snyk/view/welcome/welcomeViewProvider';
+
+suite('View Navigation', () => {
+  setup(async () => {
+    await configuration.setToken(undefined);
+    await configuration.setFeaturesConfiguration(undefined);
+  });
+
+  teardown(async () => {
+    await configuration.setToken(undefined);
+  });
+
+  test('"Feature view is seen after user authenticates within welcome view', async () => {
+    const extension = getExtension();
+    const viewContainer = extension.viewManagerService.viewContainer;
+    const welcomeTree = viewContainer.get<TreeView<Node>>(SNYK_VIEW_WELCOME);
+
+    // 1. Check welcome view is visible
+    strictEqual(welcomeTree?.visible, true);
+
+    // 2. Authenticate a user
+    await configuration.setToken('fake-token');
+
+    // Give time to pick up the setting change
+    const sleep = (duration: number) => new Promise(resolve => setTimeout(resolve, duration));
+    await sleep(1500);
```

Comment:
@michelkaporin, can it be flaky by test executions?

Reply:
@pavel-snyk I don't expect this particular to be flaky and it is not according to many runs on my machine. But I agree that time-dependant tests might fail more often, also not being huge fan of them. Do you have a better idea how can we not rely on timing here?

Reply:
🤔 at the moment I don't have a better idea how to deal with it. Let's sync offline in Slack.
=> resolved.